Build: add navbar

ceyhungulbas opened the pull request:
close #1 

Responsive:
![responsiveView](https://user-images.githubusercontent.com/44509662/129418069-34bcc21c-3c89-4888-8e9e-66da44333278.jpg)


Navbar container: 
- largeScreenNavbarData contains the data used in map func inside of Navbar.jsx
- smallScreenNavbarData contains the data used in map func inside of Navbar.jsx

hooks:
- detect the width of the screen

ceyhungulbas committed:
Build: add navbar

ceyhungulbas committed:
Build: add navbar

ceyhungulbas committed:
Test: solve useEffect problem

udenizdemirbilek reviewed (approved):
Good job! Ceyhun 👍
Right now navbar displays underneath the footer but I don't think it is a problem when the footer is at the bottom.
![Screenshot_2021-08-14-01-08-38-638_org mozilla firefox](https://user-images.githubusercontent.com/81304552/129422810-3c208ae3-ff9d-46fa-8d8e-723f285fe139.jpg)

ceyhungulbas commented:
@udenizdemirbilek Yeah yeah I saw it too. There is no content our pages now, thats why footer displaying on the top. If we have still problem when we have content, we can work on it to solve.

halitbatur commented on src/App.js:
```diff
@@ -2,11 +2,21 @@ import React from "react";
 
 function App() {
   return (
     <>
```
you can move the Footer to a route like the navbar and it should be after it

halitbatur replied:
and move everything to the Router and it can be nested inside the Container

halitbatur commented on src/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,166 @@
+  transition: 0.3s;
+}
+
+#dropdown-basic-button {
```
no ids please :)

halitbatur commented on src/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,166 @@
+    margin-left: 7%;
+  }
+
+  #dropdown-basic-button {
```
no ids please

ceyhungulbas replied:
I couldnt make it without id. I tried `className="dropdown-basic-button"` didnt worked. Also I changed the format. I used camelCase version `className="dropdownBasicButton"` . Didnt worked too.

halitbatur commented on src/components/Navbar/Navbar.jsx:
```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect, useRef } from "react";
+import "./Navbar.css";
+import "bootstrap/dist/css/bootstrap.min.css";
```
this is already improted in the app.js please remove it from here

halitbatur commented on src/components/Navbar/Navbar.jsx:
```diff
@@ -0,0 +1,154 @@
+  );
+};
+
+const NavbarForSmallScreen = () => {
```
can we move both the small screen and big screen navbar to seperate components and import them here ? I think that would be more clean

halitbatur commented on src/components/Navbar/Navbar.jsx:
```diff
@@ -0,0 +1,154 @@
+
+  useWindowDimensions();
+
+  const { width } = useWindowDimensions();
```
u only need this line of code here please delete line 143 and line 141, and the extra import for the getWindowDimensions, make sure to read the code and understand it really well, getWindowDimensions is a helper function for the useWindowDimensions hook

ceyhungulbas committed:
Build: add navbar

ceyhungulbas committed:
Test: add test

ceyhungulbas committed:
Test: yarn run fix

ceyhungulbas commented:
@halitbatur  PTAL

halitbatur commented on src/App.js:
```diff
@@ -2,13 +2,23 @@ import React from "react";
+          <Route path="/donations" component={Donations} />
+          <Route path="/about" component={About} />
+          <Route path="/contactUs" component={ContactUs} /> */}
+          <Footer />
```
can we have this in a route as well ?

halitbatur commented on src/components/Navbar/NavbarForLargeScreen.jsx:
```diff
@@ -0,0 +1,47 @@
+        </div>
+        <div className="navbarLinks">
+          {largeScreenNavbarData.map((info, index) => (
+            <Link to={info.target} className={info.cName} key={index}>
```
lets no use index as keys https://dev.to/rajnishkatharotiya/you-shouldn-t-use-an-index-as-key-of-the-element-2jf

ceyhungulbas replied:
@halitbatur Well I created specific ids for each user in data files then I changed index to info.id

halitbatur commented on src/containers/Navbar/smallScreenNavbarData.js:
```diff
@@ -0,0 +1,54 @@
+
+export const smallScreenNavbarData = [
+  {
+    cName: "navLi",
```
what does c stand for here ?

ceyhungulbas replied:
@halitbatur className, didnt want to use syntax of react so i wrote cName

halitbatur replied:
@ceyhungulbas I think u can use className here since its treated as a key

halitbatur replied:
@ceyhungulbas and since all of them have the same className why not just write it there right away instead of having it in all the objects ?

ceyhungulbas committed:
Fix: change key index to id

ceyhungulbas commented:
@halitbatur  PTAL I created specific ids for each user in data files then I changed key={index} to {info.id}

halitbatur commented on src/components/Navbar/NavbarForLargeScreen.jsx:
```diff
@@ -0,0 +1,47 @@
+              {info.name}
+            </Link>
+          ))}
+          <DropdownButton id="dropdownBasicButton" title="Language">
```
why the usage of id here ?

ceyhungulbas replied:
@halitbatur because its from react-bootstrap. I cannot change to className. It doesnt work with className

halitbatur commented on src/components/Navbar/NavbarForSmallScreen.jsx:
```diff
@@ -0,0 +1,92 @@
+              </Button>
+            </Link>
+          </li>
+
```
extra line

halitbatur commented on src/containers/Navbar/largeScreenNavbarData.js:
```diff
@@ -0,0 +1,38 @@
+  {
+    id: "d",
+    target: "/donations",
+    cName: "eachLinksInNavbar",
```
what does the c here stand for ?

ceyhungulbas committed:
Build: add navbar

ceyhungulbas commented:
@halitbatur  PTAL 

- I deleted extra line, i dont know why there is extra line, i used Format Document with Prettier extension, yarn run fix.
- I delete cName from datas and changed on map func in NavbarForSmallScreen / NavbarForLargeScreen
- Still using ID because of react-bootstrap, i tried use className didnt work.

busesisi reviewed (commented):
Nice work Ceyhun. The navbar looks neat and very close to what we had designed at the design phase. And it is functional as well. Good work, ellerine sağlık!

ceyhungulbas committed:
Fix: change id to class

ceyhungulbas commented:
@halitbatur  PTAL

- I changed ID to className
- I fixed issues on buttons
- Added hover feature on Language button
- yarn run fix
- Updated snapshots

halitbatur commented on src/components/Navbar/__tests__/NavbarForLargeScreen.test.js:
```diff
@@ -0,0 +1,11 @@
+
+afterEach(cleanup);
+
+it("matches snapshot", () => {
```
please change test name

ceyhungulbas replied:
https://github.com/ReCoded-Org/capstone-turkey-giftify/pull/26/commits/71458818b1c9fcd0f37fffea28e6406da9de5d89

I changed test name like 1 hour ago

ceyhungulbas committed:
Test: change test message

halitbatur commented on src/containers/Navbar/largeScreenNavbarData.js:
```diff
@@ -0,0 +1,32 @@
+    target: "/",
+    name: "Home",
+  },
+  {
```
can we move these data files into a seprate folder and call it data ? instead of it being in the container

halitbatur commented:
@ceyhungulbas just move the data files and its ready to be merged

ceyhungulbas committed:
Build: add navbar

ceyhungulbas committed:
Test: fix typo
